types/hastscript-tests.ts

```diff
@@ -0,0 +1,5 @@
+import h = require('hastscript')
+
+h()
+h('.bar', {class: 'bar'})
+h('.foo', {class: 'bar'}, h('.baz'))
```

Comment:
Could we add some additional test cases for
* text and array children
* properties being skipped
* return type
* a case were typing should reject

For example:

```suggestion
h() // $ExpectType Element
h('.bar', {class: 'bar'}) // $ExpectType Element
h('.bar', 'child text') // $ExpectType Element
h('.bar', ['child text']) // $ExpectType Element
h('.foo', {class: 'bar'}, h('.baz')) // $ExpectType Element
h('.foo', {class: 'bar'}, [h('.baz')]) // $ExpectType Element
h('.bar',{class: 'bar'} , 'child text') // $ExpectType Element
h('.bar',{class: 'bar'}, ['child text']) // $ExpectType Element
h(false) // $ExpectError
```